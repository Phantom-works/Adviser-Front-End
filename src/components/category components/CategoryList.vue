<template>
  <div v-for="category in categories" :key="category.id">
    <GameCategory :category="category"/>
  </div>
</template>

<script setup lang = ts>

import type Category from "@/components/Interfaces/Category";
import GameCategory from "@/components/category components/GameCategory.vue";

defineProps<{categories: Category[]}>()
</script>

<style scoped>

</style>

