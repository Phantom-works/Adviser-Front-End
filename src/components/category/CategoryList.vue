<template>
  <div v-for="category in categories" :key="category.id" class="categories-container">
    <CoCategory :category="category" @DeleteCategory="DeleteCategory"/>
  </div>
</template>

<script lang="ts" setup>
import type Category from '@/Model/Category'
import CoCategory from '@/components/category/CategoryBubble.vue'

defineProps<{ categories: Category[] }>()

const emit = defineEmits(['DeleteCategory'])
function DeleteCategory (_id: number) {
  emit("DeleteCategory", _id)
}
</script>

<style scoped>
.categories-container {
  display: inline-block;
}
</style>
