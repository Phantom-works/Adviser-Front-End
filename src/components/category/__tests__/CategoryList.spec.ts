import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import CategoryList from '../CategoryList.vue'
import type Category from '../../../Model/Category'
describe('CategoryList Snapshots', () => {
  // Arrange

  it('Snapshot Empty List', () => {
    //Arrange
    const categories = <Category[]>[]

    // Act
    const wrapper = mount(CategoryList, {
      props: {
        categories: categories
      }
    })
    // Assert
    expect(wrapper.html()).toMatchSnapshot()
  })
})
