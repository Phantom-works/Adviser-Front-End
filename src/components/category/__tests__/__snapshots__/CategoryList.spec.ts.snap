// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CategoryList Snapshots > Snapshot 9 categories differentiating text length 1`] = `
"<div data-v-43684615=\\"\\" class=\\"categories-container\\">
  <div data-v-577a7027=\\"\\" data-v-43684615=\\"\\" class=\\"Bubble\\">
    <h3 data-v-577a7027=\\"\\">CategoRUEYJTWRHGQEYJEYHNTRAGEVry 1 <i data-v-577a7027=\\"\\" class=\\"fa-solid fa-x\\"></i></h3>
  </div>
</div>
<div data-v-43684615=\\"\\" class=\\"categories-container\\">
  <div data-v-577a7027=\\"\\" data-v-43684615=\\"\\" class=\\"Bubble\\">
    <h3 data-v-577a7027=\\"\\">Category 2 <i data-v-577a7027=\\"\\" class=\\"fa-solid fa-x\\"></i></h3>
  </div>
</div>
<div data-v-43684615=\\"\\" class=\\"categories-container\\">
  <div data-v-577a7027=\\"\\" data-v-43684615=\\"\\" class=\\"Bubble\\">
    <h3 data-v-577a7027=\\"\\">ry 3 <i data-v-577a7027=\\"\\" class=\\"fa-solid fa-x\\"></i></h3>
  </div>
</div>
<div data-v-43684615=\\"\\" class=\\"categories-container\\">
  <div data-v-577a7027=\\"\\" data-v-43684615=\\"\\" class=\\"Bubble\\">
    <h3 data-v-577a7027=\\"\\">CategoEYHTWRAGEfaqathbwyjnhtarqEGry 1 <i data-v-577a7027=\\"\\" class=\\"fa-solid fa-x\\"></i></h3>
  </div>
</div>
<div data-v-43684615=\\"\\" class=\\"categories-container\\">
  <div data-v-577a7027=\\"\\" data-v-43684615=\\"\\" class=\\"Bubble\\">
    <h3 data-v-577a7027=\\"\\">Category 2 <i data-v-577a7027=\\"\\" class=\\"fa-solid fa-x\\"></i></h3>
  </div>
</div>
<div data-v-43684615=\\"\\" class=\\"categories-container\\">
  <div data-v-577a7027=\\"\\" data-v-43684615=\\"\\" class=\\"Bubble\\">
    <h3 data-v-577a7027=\\"\\">CatHWTGQRTH6WGQREgthwtgaqreegory 3 <i data-v-577a7027=\\"\\" class=\\"fa-solid fa-x\\"></i></h3>
  </div>
</div>
<div data-v-43684615=\\"\\" class=\\"categories-container\\">
  <div data-v-577a7027=\\"\\" data-v-43684615=\\"\\" class=\\"Bubble\\">
    <h3 data-v-577a7027=\\"\\">Categ4HJY7YH5WT6G4R5WHJ5YWTGRQEFGTHYJW5TGRQE4HWory 1 <i data-v-577a7027=\\"\\" class=\\"fa-solid fa-x\\"></i></h3>
  </div>
</div>
<div data-v-43684615=\\"\\" class=\\"categories-container\\">
  <div data-v-577a7027=\\"\\" data-v-43684615=\\"\\" class=\\"Bubble\\">
    <h3 data-v-577a7027=\\"\\">Catego <i data-v-577a7027=\\"\\" class=\\"fa-solid fa-x\\"></i></h3>
  </div>
</div>
<div data-v-43684615=\\"\\" class=\\"categories-container\\">
  <div data-v-577a7027=\\"\\" data-v-43684615=\\"\\" class=\\"Bubble\\">
    <h3 data-v-577a7027=\\"\\">Cat <i data-v-577a7027=\\"\\" class=\\"fa-solid fa-x\\"></i></h3>
  </div>
</div>"
`;

exports[`CategoryList Snapshots > Snapshot Empty List 1`] = `""`;

exports[`CategoryList Snapshots > Snapshot no categories 1`] = `""`;

exports[`CategoryList Snapshots > Snapshot text empty 3 categories 1`] = `
"<div data-v-43684615=\\"\\" class=\\"categories-container\\">
  <div data-v-577a7027=\\"\\" data-v-43684615=\\"\\" class=\\"Bubble\\">
    <h3 data-v-577a7027=\\"\\"> <i data-v-577a7027=\\"\\" class=\\"fa-solid fa-x\\"></i></h3>
  </div>
</div>
<div data-v-43684615=\\"\\" class=\\"categories-container\\">
  <div data-v-577a7027=\\"\\" data-v-43684615=\\"\\" class=\\"Bubble\\">
    <h3 data-v-577a7027=\\"\\"> <i data-v-577a7027=\\"\\" class=\\"fa-solid fa-x\\"></i></h3>
  </div>
</div>
<div data-v-43684615=\\"\\" class=\\"categories-container\\">
  <div data-v-577a7027=\\"\\" data-v-43684615=\\"\\" class=\\"Bubble\\">
    <h3 data-v-577a7027=\\"\\"> <i data-v-577a7027=\\"\\" class=\\"fa-solid fa-x\\"></i></h3>
  </div>
</div>"
`;

exports[`CategoryList Snapshots > Snapshot text long 5 categoeries 1`] = `
"<div data-v-43684615=\\"\\" class=\\"categories-container\\">
  <div data-v-577a7027=\\"\\" data-v-43684615=\\"\\" class=\\"Bubble\\">
    <h3 data-v-577a7027=\\"\\">Category 1rmyumednthsbrg <i data-v-577a7027=\\"\\" class=\\"fa-solid fa-x\\"></i></h3>
  </div>
</div>
<div data-v-43684615=\\"\\" class=\\"categories-container\\">
  <div data-v-577a7027=\\"\\" data-v-43684615=\\"\\" class=\\"Bubble\\">
    <h3 data-v-577a7027=\\"\\">Category 2muenytwhsrgbavm4eunwty <i data-v-577a7027=\\"\\" class=\\"fa-solid fa-x\\"></i></h3>
  </div>
</div>
<div data-v-43684615=\\"\\" class=\\"categories-container\\">
  <div data-v-577a7027=\\"\\" data-v-43684615=\\"\\" class=\\"Bubble\\">
    <h3 data-v-577a7027=\\"\\">Category 3muerynthbrwsegahnjuermjtywsrhbaegvFWED <i data-v-577a7027=\\"\\" class=\\"fa-solid fa-x\\"></i></h3>
  </div>
</div>
<div data-v-43684615=\\"\\" class=\\"categories-container\\">
  <div data-v-577a7027=\\"\\" data-v-43684615=\\"\\" class=\\"Bubble\\">
    <h3 data-v-577a7027=\\"\\">Category 1entyawrtegTHJWKE6U8JYHWRrgat6ysj7iksrthaergwefrghatES6UJ <i data-v-577a7027=\\"\\" class=\\"fa-solid fa-x\\"></i></h3>
  </div>
</div>
<div data-v-43684615=\\"\\" class=\\"categories-container\\">
  <div data-v-577a7027=\\"\\" data-v-43684615=\\"\\" class=\\"Bubble\\">
    <h3 data-v-577a7027=\\"\\">Category 2KEJYHRTAEGFWDMUJNTYHRGB <i data-v-577a7027=\\"\\" class=\\"fa-solid fa-x\\"></i></h3>
  </div>
</div>"
`;
